src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -173,27 +173,33 @@ suggestRemoveRedundantImport ParsedModule{pm_parsed_source = L _  HsModule{hsmod
         = [("Remove import", [TextEdit (extendToWholeLineIfPossible contents _range) ""])]
     | otherwise = []
 
-suggestDeleteTopBinding :: ParsedModule -> Diagnostic -> [(T.Text, [TextEdit])]
-suggestDeleteTopBinding ParsedModule{pm_parsed_source = L _ HsModule{hsmodDecls}} Diagnostic{_range=_range,..}
+suggestDeleteTopBinding :: Maybe T.Text -> ParsedModule -> Diagnostic -> [(T.Text, [TextEdit])]
+suggestDeleteTopBinding sourceOpt ParsedModule{pm_parsed_source = L _ HsModule{hsmodDecls}} Diagnostic{_range=_range,..}
 -- Foo.hs:4:1: warning: [-Wunused-top-binds] Defined but not used: ‘f’
     | Just [name] <- matchRegex _message ".*Defined but not used: ‘([^ ]+)’"
-    , let allTopLevel = filter (isTopLevel . fst)
+    , let allTopLevel = filter ((== 0) . (_character . _start) . fst)
```

Comment:
I would bring back the `isTopLevel` helper

Reply:
While this may work for most haskell files out in the wild, it is not strictly correct. For instance, the following is a valid haskell module:

```haskell
module Main (main) where
  main :: IO ()
  main = putStrLn "hello"

```

(I understand this bug was not introduced by this PR)